Framework/DataHandling/src/GenerateGroupingPowder.cpp

```diff
@@ -70,52 +278,94 @@ void GenerateGroupingPowder::exec() {
   if (detectorIDs.empty())
     throw std::invalid_argument("Workspace contains no detectors.");
 
+  const auto inst = input_ws->getInstrument();
+  if (!inst) { // validator should make this impossible
+    throw std::invalid_argument("Input Workspace has invalid Instrument\n");
+  }
+
+  auto groupWS = std::make_shared<GroupingWorkspace>(inst);
+  this->setProperty("GroupingWorkspace", groupWS);
+
   const double step = getProperty("AngleStep");
-  const auto numSteps = static_cast<size_t>(180. / step + 1);
 
-  std::vector<std::vector<detid_t>> groups(numSteps);
-  std::vector<double> twoThetaAverage(numSteps, 0.);
-  std::vector<double> rAverage(numSteps, 0.);
+  Labelor *label;
```

Comment:
```suggestion
  std::unique_ptr<Labelor> label;
```
There's a memory leak if line 331 throws.

Reply:
fixed